src/Microsoft.Azure.SignalR.Common/Auth/JwtBuilders.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Buffers;
+using System.Buffers.Text;
+using System.Diagnostics;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text.Json;
+using System.Security.Claims;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.IdentityModel.Logging;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class JwtBuilder
+    {
+        // Registered claims
+        public static string Nbf => "nbf";
+        public static string Exp => "exp";
+        public static string Iat => "iat";
+        public static string Aud => "aud";
+        public static string Sub => "sub";
+        public static string Iss => "iss";
+        public static string Jti => "jti";
+
+        private readonly byte[] _key;
+        private readonly JwtData _jwtHeader;
+        private readonly JwtData _jwtPayload;
+        private readonly AccessTokenAlgorithm _algorithm;
+
+        public JwtBuilder(
+            DateTime? notBefore = null, 
+            DateTime? expires = null, 
+            DateTime? issuedAt = null, 
+            string issuer = null, 
+            string audience = null, 
+            IEnumerable<Claim> claims = null, 
+            byte[] key = null, 
+            string kid = null, 
+            AccessTokenAlgorithm algorithm = AccessTokenAlgorithm.HS256)
+        {
+            _key = key;
+            _algorithm = algorithm;
+            _jwtHeader = GenerateJwtHeader(kid, _algorithm);
+            _jwtPayload = GenerateJwtPayload(issuer, audience, claims, notBefore, expires, issuedAt);
+
+        }
+
+        /// <summary>
+        /// Create a <see cref="JwtData"/> representing JWT payload according to registered claims and <paramref name="claims"/> which represents a series of claims.
+        /// </summary>
+        /// <returns>a <see cref="JwtData"/> representing JWT payload</returns>
+        private JwtData GenerateJwtPayload(string issuer = null, string audience = null, IEnumerable<Claim> claims = null, DateTime ?notBefore = null, DateTime ?expires = null, DateTime ?issuedAt = null)
+        {
+            JwtData jwtPayload = new JwtData();
+
+            var subject = claims == null ? null : new ClaimsIdentity(claims);
+            // add claims
+            jwtPayload.AddClaims(claims);
```

Comment:
thinking: shall we still support shortening the supported claim types?

Reply:
Otherwise currently working claims might exceed our length limit (1024) after the change

Reply:
fixed